cmd/opera/launcher/validator.go

```diff
@@ -33,38 +33,28 @@ var validatorPasswordFlag = cli.StringFlag{
 // command line flags or from the keystore if CLI indexed.
 func setValidator(ctx *cli.Context, cfg *emitter.Config) error {
```

Comment:
should not we call error check logic at the beginning of `setValidator`  or ? 
```
func setValidator(ctx *cli.Context, cfg *emitter.Config) error {
      	if cfg.Validator.ID != 0 && cfg.Validator.PubKey.Empty() {
		      return errors.New("validator public key is not set")
	} 
    ....
   
}

Reply:
`setValidator` method updates `cfg.Validator`. `cfg.Validator` may be (and normally is) empty before calling `setValidator`
The goal of the check is to enforce that "if validator ID is set, then validator pubkey must be set too"